pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -326,54 +365,93 @@ func (r *ReconcileAWSFederatedAccountAccess) createIAMRole(awsClient awsclient.C
 	// Marshal assumeRolePolicyDoc to json
 	jsonAssumeRolePolicyDoc, err := json.Marshal(&assumeRolePolicyDoc)
 	if err != nil {
-		return nil, fmt.Errorf("Error marshalling jsonPolicy doc : Error %s", err.Error())
+		return nil, err
+	}
+
+	var roleName string
+	// Try and build role name
+	if uuidLabel, ok := afaa.Labels["uuid"]; ok {
+		roleName = afr.Name + "-" + uuidLabel
```

Comment:
Similar to above, do we need to do any checks here to make sure that that the `afr.Spec.AWSCustomPolicy.Name` is not longer than 64 characters minus the 37 characters from the `-[uuid[?

Maybe even shorter? (Are we using the Switch Role feature or are we assuming this role through another mechanism?)

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-limits.html#w477aac29b9c11b5

Reply:
Probably same as above, we should validate length before trying to create. We're using switch role since we put the url in the CR

Reply:
This was a mistake on my part where I missed that UUID was only 6 random characters and not a full 36 char UUID string.  We should still add a check here for length, but it's definitely not as bad now with us only using up 7 out of 64 chars.